testcases/veryl/55_generic_module.veryl

```diff
@@ -41,3 +42,9 @@ module Module55H::<W: const> {
 
     let _a: StructH::<W> = 0;
 }
+
```

Comment:
I think this example is bit strange.
`IF` is bound by `Interface55I`, but `Interface55I` is not `proto` but specific type, so `IF` is always `Interface55I`.
Therefore I think generics is not required in this case.
How about adding `proto interface` support described at #963 like below?

```systemverilog
proto interface PInterface {
    function FuncA(a: input logic, b: input logic) -> logic;
}

interface Interface55I for PInterface {
    function FuncA(a: input logic, b: input logic) -> logic {
        return 0;
    }
}

module Module55I::<IF: PInterface> {
    inst u: IF;
}

module X {
    inst u: Module55I::<Interface55I>;
}
```

Reply:
I agree with you but a lot of changes is needed to introduce `proto interface`.
For now, how about removing this example and opening a new PR to introduce `proto interface`?

Reply:
I agree you. `check_type.rs:387-390` and `symbol.rs:304` should be removed probably.

Reply:
I removed changes to allow interface to be used as generic boundary.